### PR TITLE
Add support for SOCKS proxy through 'socks_proxy' environment variable

### PR DESCRIPTION
**Pull request by NZSmartie**
This PR is mostly intended for the [Bitwarden CLI](/bitwarden/cli). Adding support for a SOCKS4 or SOCKS5 proxy by using the `socks-proxy-agent` library. 

Since jslib already supports HTTP or HTTPS proxies through the `http_proxy` or `https_proxy` environment variables respectively, it made sense to also support SOCKS proxies. The SOCKS proxy library is provided by the same author being `https-proxy-agent`.

Example usage:
```bash
socks_proxy=socks5h://127.0.01:8010 bw login
```

### CODE REVIEW

**Comment by NZSmartie** [comment]
Woops. I have LTS of Node.JS installed and worked on my dev machine, i'll get v10 LTS installed and fix these errors.

File: tsconfig.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "emitDecoratorMetadata": true,
     "declarationDir": "dist/types",
     "outDir": "dist",
+    "skipLibCheck": true,
```
Comment (NZSmartie) [inline tsconfig.json:16]: This is a "dirty hack" to get this Pull Request to compile. I'm not sure why this change will compile in `v12.17.0` but not in `v10.20.1` without telling typescript to skip checking libs in `node_modules`

**Comment by Hinton** [comment]
Hi,

We are currently in the process of archiving this repository and moving the code and future development to the [bitwarden/clients](https://github.com/bitwarden/clients) repository. This PR will be closed but feel free to re-open it again on [bitwarden/clients](https://github.com/bitwarden/clients).
